test/successful-directories.test.js

```diff
@@ -156,6 +156,52 @@ describe('cosmiconfig', () => {
       ]);
     });
 
+    testSyncAndAsync(
+      'finds JS ES Modules file in first searched dir',
+      sync => () => {
+        function readFile(searchPath) {
+          switch (searchPath) {
+            case absolutePath('a/b/c/d/e/f/package.json'):
+            case absolutePath('a/b/c/d/e/f/.foorc'):
+            case absolutePath('a/b/c/d/e/package.json'):
+            case absolutePath('a/b/c/d/e/.foorc'):
+            case absolutePath('a/b/c/d/e/foo.config.js'):
+              throw { code: 'ENOENT' };
+            case absolutePath('a/b/c/d/e/f/foo.config.js'):
+              return `
+Object.defineProperty(exports, '__esModule', {
```

Comment:
I *would* suggest shoving that text down to the bottom in a function e.g.

```js
getFooConfigJs() {
  return `
...
`
}
```

but there's probably a better way - so I *do* suggest to hang tight til davidtheclark or sudo-suhas gets around to recommending something better ha.

Reply:
I am not sure I understand what you are suggesting here. I copied the format from https://github.com/davidtheclark/cosmiconfig/blob/44a65ab84dd223b231e55e6e3752b8736564d872/test/successful-directories.test.js#L134

Reply:
yeah that makes sense.  For some reason the 8 lines of template text seems to get in the way for me - but that's just preference and mine will probably conflict with what the other guys think.  No worries

Reply:
I made it two lines. Better?